src/fabric/src/fabric2_fdb.erl

```diff
@@ -1066,26 +1110,45 @@ local_doc_to_fdb(Db, #doc{} = Doc) ->
         body = Body
     } = Doc,
 
+    Key = erlfdb_tuple:pack({?DB_LOCAL_DOCS, Id}, DbPrefix),
+
     StoreRev = case Rev of
         _ when is_integer(Rev) -> integer_to_binary(Rev);
         _ when is_binary(Rev) -> Rev
     end,
 
-    Key = erlfdb_tuple:pack({?DB_LOCAL_DOCS, Id}, DbPrefix),
-    Val = {StoreRev, Body},
-    {Key, term_to_binary(Val, [{minor_version, 1}])}.
+    BVal = term_to_binary(Body, [{minor_version, 1}]),
+    {Rows, _} = lists:mapfoldl(fun(Chunk, ChunkId) ->
+        K = erlfdb_tuple:pack({?DB_LOCAL_DOC_BODIES, Id, ChunkId}, DbPrefix),
+        {{K, Chunk}, ChunkId + 1}
+    end, 0, chunkify_binary(BVal)),
 
+    {Key, StoreRev, Rows}.
 
-fdb_to_local_doc(_Db, DocId, Bin) when is_binary(Bin) ->
-    {Rev, Body} = binary_to_term(Bin, [safe]),
+
+fdb_to_local_doc(_Db, DocId, Val, []) when is_binary(Val) ->
```

Comment:
The subtlety of the empty array here as a check for the upgrade is a bit too loose I think. Adding a check for the `<<131, _/binary>>` on `Val` would help. I can't think of a way that the empty list might happen but I can't convince myself it's not impossible.

Reply:
Good idea